packages/modules/cache/src/utils/strategy.ts

```diff
@@ -0,0 +1,159 @@
+import {
+  type GraphQLSchema,
+  Modules,
+  toCamelCase,
+  upperCaseFirst,
+} from "@medusajs/framework/utils"
+import type { Event, ICachingModuleService } from "@medusajs/framework/types"
+import type { InjectedDependencies } from "@types"
+import crypto from "crypto"
+import stringify from "fast-json-stable-stringify"
+import { CacheInvalidationParser, EntityReference } from "./parser"
+import { type CachingModuleService } from "@services"
+
+export function objectHash(input: any): string {
+  const str = stringify(input)
+  return crypto.createHash("sha1").update(str).digest("hex")
+}
+
+export class DefaultCacheStrategy {
+  #cacheInvalidationParser: CacheInvalidationParser
+  #container: InjectedDependencies
+  #cacheModule: ICachingModuleService
+
+  constructor(
+    container: InjectedDependencies,
+    schema: GraphQLSchema,
+    cacheModule: CachingModuleService
+  ) {
+    this.#cacheInvalidationParser = new CacheInvalidationParser(schema)
+    this.#container = container
+    this.#cacheModule = cacheModule
+
+    this.applyReactiveInvalidation()
+  }
+
+  applyReactiveInvalidation() {
+    const eventBus = this.#container[Modules.EVENT_BUS]
+
```

Comment:
self-review: as discussed with carlos, this caching could expose a method to publish messages and the event bus if the cache is available could provide the events to it so that they get published as messages (pub/sub without bullmq) through the opened stream and not consumed the lua execution limit. then caching can just rely on messages to invalidate